### PR TITLE
Add ISO 3166-1 data accessors to the Country model

### PR DESCRIPTION
**Pull request by damsfx**

### CODE REVIEW

**Comment by LukeTowers** [comment]
Can you add some tests and documentation to the readme?

File: README.md [hunk 1]
```diff
@@ -35,82 +39,123 @@ This will automatically create two "belongs to" relationships:
...
+// Softly looks up and sets the country_id and state_id
+// for these Country and State relations.
+
+$model->country_code = "US";
+$model->state_code = "FL";
+$model->save();
+```
+### ISO 3166-1 accessors
+
+The behavior will also provide specials ISO-3166-1 accessors to the model through the [league/iso3166](https://github.com/thephpleague/iso3166).
```
Comment (LukeTowers) [inline README.md:129]:
```suggestion
The behavior will also provide specials ISO-3166-1 accessors to the model through the [league/iso3166](https://github.com/thephpleague/iso3166) package.  
```

**Comment by LukeTowers** [comment]
@damsfx let me know if you're able to add tests or not, I'm fine merging either way at this point.

**Comment by damsfx** [comment]
> @damsfx let me know if you're able to add tests or not, I'm fine merging either way at this point.

Sorry @LukeTowers, I've had a crazy day and haven't had time to finish integrating the tests.  
I'm going to try to find some time to finalise this PR.

File: tests/Feature/LocationIsoTest.php [hunk 2]
```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace Winter\Location\Tests\Feature;
+
+use Winter\Location\Models\Country;
+use System\Tests\Bootstrap\PluginTestCase;
+
+class LocationIsoTest extends PluginTestCase
+{
+
```
Comment (LukeTowers) [inline tests/Feature/LocationIsoTest.php:10]:
```suggestion
```

File: tests/Feature/LocationIsoTest.php [hunk 3]
```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace Winter\Location\Tests\Feature;
+
+use Winter\Location\Models\Country;
+use System\Tests\Bootstrap\PluginTestCase;
```
Comment (LukeTowers) [inline tests/Feature/LocationIsoTest.php:6]:
```suggestion
use System\Tests\Bootstrap\PluginTestCase;
use Winter\Location\Models\Country;
```

File: tests/Unit/LocationModelsTest.php [hunk 4]
```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace Winter\Location\Tests\Unit;
+
+use Winter\Location\Models\Country;
+use Winter\Location\Models\State;
+use System\Tests\Bootstrap\PluginTestCase;
```
Comment (LukeTowers) [inline tests/Unit/LocationModelsTest.php:7]:
```suggestion
use System\Tests\Bootstrap\PluginTestCase;
use Winter\Location\Models\Country;
use Winter\Location\Models\State;
```

File: tests/Unit/LocationPluginTest.php [hunk 5]
```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace Winter\Location\Tests\Unit;
+
+use Winter\Location\Plugin;
+use System\Classes\PluginBase;
+use System\Tests\Bootstrap\PluginTestCase;
```
Comment (LukeTowers) [inline tests/Unit/LocationPluginTest.php:7]:
```suggestion
use System\Classes\PluginBase;
use System\Tests\Bootstrap\PluginTestCase;
use Winter\Location\Plugin;
```

**Comment by LukeTowers** [comment]
@damsfx can you add the github workflows files from https://github.com/wintercms/wn-translate-plugin/tree/main/.github/workflows so that we can also have them run on this repository?

**Comment by damsfx** [comment]
> @damsfx can you add the github workflows files from https://github.com/wintercms/wn-translate-plugin/tree/main/.github/workflows so that we can also have them run on this repository?

Do I need to adapt the tests to take the 1.x branch of Winter into account, using the same logic as in this file in this case?
[Winter.Translate TranslatePluginTestCase](https://github.com/wintercms/wn-translate-plugin/blob/main/tests/TranslatePluginTestCase.php)

**Comment by LukeTowers** [comment]
@damsfx meh, I'll leave that up to you. This plugin is unlikely to need to be updated for any sites / projects still on 1.0 so I don't think we need to provide tests for that old of a version.

**Comment by damsfx** [comment]
@LukeTowers I've tried to set up the workflow, but I'm running into a problem.  
As the plugin requires an external dependency, it's not possible to simply copy the plugin into the Winter installation. 
https://github.com/damsfx/wn-location-plugin/blob/a2661e111075f7cb639fd1460cf02ba2fa37f068/.github/workflows/tests.yml#L61-L62

What do you think is the best way to install `league/iso3166` dependency in tests?
- Add an extra step after `Install Composer dependencies` to install plugin dependencies
- Add an extra step before `Install Composer dependencies` to modify the `composer.json` file (either to modify the `require` or `extra/merge-plgin/include` section)

**Comment by damsfx** [comment]
@LukeTowers Finally found a solution by allowing all plugin dependencies in `merge-plugin`.

```yml
      - name: Allow merge for plugins dependencies
        run: sed -i 's/myauthor/\*/g' composer.json
```

**Comment by LukeTowers** [comment]
@damsfx not a terrible workaround, although I could see it causing issues if we were testing a plugin that required other plugins. Another solution could be to run composer install from inside the plugin directory, but I'm fine with this approach for now.

Thanks!

File: README.md [hunk 6]
```diff
@@ -35,82 +39,123 @@ This will automatically create two "belongs to" relationships:
...
+// Softly looks up and sets the country_id and state_id
+// for these Country and State relations.
+
+$model->country_code = "US";
+$model->state_code = "FL";
+$model->save();
+```
+### ISO 3166-1 accessors
+
+The behavior will also provide specials ISO-3166-1 accessors to the model through the [league/iso3166](https://github.com/thephpleague/iso3166) package.
```
Comment (LukeTowers) [inline README.md:129]:
```suggestion
The behavior will also add the ISO-3166-1 values as accessors to the model (data sourced from the [league/iso3166](https://github.com/thephpleague/iso3166) package).  
```

**Comment by damsfx** [comment]
@LukeTowers 
```yml
      - name: Allow merge for plugins dependencies
        run: sed -i 's/myauthor\/\*/winter\/location/g' composer.json
```
In this case, only this plugin dependencies are allowed.

**Pull request merged**